src/porepy/compositional/materials.py

```diff
@@ -0,0 +1,543 @@
+"""Storage classes for solid constants and fluid components.
+
+Material contants are values representing either constant physical properties (e.g.
+critical pressurre) or parameters for constitutive laws (e.g. constant compressibility
+for exponential-type density law). A material is instantiated with a
+:class:`~porepy.models.units.Units` object, which defines the units of the physical
+properties for a simulation setup. While the constants must be given in base SI units
+when instantiating a material class, its constants are converted and stored in the
+target units, to be used subsequently.
+
+For converting values on the fly, see :meth:`~porepy.models.units.Units.convert_units`.
+
+"""
+
+from __future__ import annotations
+
+from dataclasses import FrozenInstanceError, asdict, dataclass, field, is_dataclass
+from typing import Any, Callable, ClassVar, TypeVar, cast
+
+import porepy as pp
+
+from .base import Component
+
+__all__ = [
+    "Constants",
+    "FluidComponent",
+    "SolidConstants",
+    "ReferenceVariableValues",
+    "load_fluid_constants",
+]
+
+number = pp.number
+
+
+# 1. By using keyword_only arguments for the construction of materials, the user is
+# forced to instantiate the constants with the right names of constants, otherwise
+# errors are raised (no separate checks required).
+# 2. By providing default values to the fields, not every constant is required. The user
+# is expected to be aware which physics are used in the model.
+# 3. By instructing dataclass to not override the __eq__ provided by object (we do not
```

Comment:
Please fix sentence. The by suggests a "By [action], [something results]" structure.

Reply:
Done.